visualiser/app.py

```diff
@@ -0,0 +1,212 @@
+from dash import dcc, Input, Output, callback, html, dash_table
+from plotly.subplots import make_subplots
+from rich.console import Console
+from pprint import pprint
+from decimal import Decimal
+
+import os
+import sys
+import dash
+import dash_bootstrap_components as dbc
+import pandas as pd
+import plotly.graph_objects as go
+
+console = Console()
+
+args = sys.argv[1:]
+
+if len(args) > 0:
+    logsdir = args[0]
+else:
+    logsdir = "log_folder"
+    console.print(f"No <logs_dir> argument given. Using 'log_folder'.", style="bold white")
+
+if not os.path.exists(logsdir):
+    console.print(f"{logsdir} is NOT a valid path.", style="bold red")
+    sys.exit(0)
+
+try:
+    op_df = pd.read_csv(os.path.join(logsdir, "operations.csv"), index_col=False)
+except FileNotFoundError as e:
+    op_df = pd.read_csv(os.path.join(logsdir, "operation.csv"), index_col=False)
+    
+def truncate_string(s, length):
+    if isinstance(s, str) and len(s) > length:
+        return s[:length] + "..."
+    else:
+        return s
+    
+op_df["src_truncated"] = op_df['src'].apply(truncate_string, length = 5)
```

Comment:
This will destroy the distribution of peer IDs (this is important for us!). Rather create a new dictionary with mapping virtual IDs to real IDs (and potentially vice versa). 
```
if real_id not in mapping:
     mapping[virtual_id++] = real_id
    
```